idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt

```diff
@@ -40,6 +40,7 @@ class RedundantCompanionReferenceInspection : AbstractKotlinInspection() {
             if (expression.text != objectDeclaration.name) return
 
             val containingClass = objectDeclaration.containingClass() ?: return
+            if (parent.getStrictParentOfType<KtEnumEntry>()?.containingClass() == containingClass) return
```

Comment:
Self-referencing our own `companion object` is always redundant, similar to how we never qualify static methods when inside the class that declares it.